service_packs/kubernetes/pod_security_policy/psp.go

```diff
@@ -0,0 +1,273 @@
+	}
+
+	if value != "not have a value provided" {
+		useValue = true
+		boolValue, err = strconv.ParseBool(value)
+		if err != nil {
+			return utils.ReformatError("Expected 'true' or 'false' but found '%s'", value) // No payload is necessary if an invalid value was provided
+		}
+	}
+
+	stepTrace.WriteString(fmt.Sprintf("Build a pod spec with default values; "))
+	securityContext := constructors.DefaultContainerSecurityContext()
+	podObject := constructors.PodSpec(Probe.Name(), config.Vars.ServicePacks.Kubernetes.ProbeNamespace, securityContext)
+	//TODO: Unit test that this always is true: len(podObject.Spec.Containers) > 0
+
+	if useValue {
+		stepTrace.WriteString(fmt.Sprintf("Set '%v' to '%v' in pod spec; ", key, value))
+		switch key {
+		case "allowPrivilegeEscalation":
+			podObject.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation = &boolValue
+		default:
+			return utils.ReformatError("Unsupported key provided: %s", key) // No payload is necessary if an invalid key was provided
+		}
+	}
+
+	stepTrace.WriteString(fmt.Sprintf("Create pod from spec; "))
+	createdPodObject, creationErr := scenario.createPodfromObject(podObject)
+
+	stepTrace.WriteString(fmt.Sprintf("Validate pod creation %s; ", result))
+
+	// Leaving these checks verbose for clarity
+	switch shouldCreate {
+	case true:
+		if creationErr != nil {
+			err = utils.ReformatError("Pod creation did not succeed: %v", creationErr)
+		}
+	case false:
+		if creationErr == nil {
+			err = utils.ReformatError("Pod creation succeeded, but should have failed")
+		} else {
+			if !errors.IsStatusCode(403, creationErr) {
+				err = utils.ReformatError("Unexpected error during Pod creation : %v", creationErr)
+			}
+		}
+	}
+
+	payload = struct {
+		RequestedPod  *apiv1.Pod
+		CreatedPod    *apiv1.Pod
+		CreationError error
+	}{
+		RequestedPod:  podObject,
+		CreatedPod:    createdPodObject,
+		CreationError: creationErr,
+	}
+
+	return err
+}
+
+func (scenario *scenarioState) theExecutionOfAXCommandInsideThePodIsY(permission, result string) error {
+	// Supported permissions:
+	//     'non-privileged'
+	//     'privileged'
+	//
+	// Supported results:
+	//     'successful'
+	//     'rejected'
+
+	stepTrace, payload, err := utils.AuditPlaceholders()
+	defer func() {
+		scenario.audit.AuditScenarioStep(stepTrace.String(), payload, err)
+	}()
+
+	// Guard clause
+	if len(scenario.pods) == 0 {
+		err = utils.ReformatError("Pod failed to create in the previous step")
+		return err
+	}
+
+	var cmd string
+	switch permission {
+	case "non-privileged":
+		cmd = "ls"
+	case "privileged":
+		cmd = "sudo ls"
+	}
+
+	var expectedExitCode int
+	switch result {
+	case "successful":
+		expectedExitCode = 0
+	case "rejected":
+		expectedExitCode = 126 // If a command is found but is not executable, the return status is 126
+		// Known issue: we can't guarantee that the 126 recieved by kubectl isn't a masked 127
+	}
```

Comment:
Same as above. For protection, should we add a default clause here and error?
```suggestion
default:
return utils.ReformatError("Unexpected value provided for expected result: %s", result)
}
```
If invalid value is accidentally passed, execution would continue with `expectedExitCode=0`